components/cmake/find_dep_packages.cmake

```diff
@@ -0,0 +1,36 @@
+# This file is for finding pacakges needed by E3SM. It should be included
+# from the main CMakeLists.txt file.
+#
+# Finding the correct packages will likely depend on the ${Package}_ROOT
+# environment variable being set by config_machines.xml for your machine. Note
+# that is environment var is case sensitive.
+
+# Machine env vars should follow the following pattern:
+#   <env name="Package_ROOT">$SHELL{if [ -z "$Package_ROOT" ]; then echo /default/install/location; else echo "$Package_ROOT"; fi}</env>
+#
+# This will allow users to easily specify a different location for all their cases by
+# simply setting ${Package}_ROOT in their shell.
+
+# Kokkos' find_package needs to come before other find_packages
+# that may define Kokkos targets so we can avoid duplicate target
+# errors.
+if (USE_KOKKOS)
+
+  # Kokkos will be built in the sharedlibs if Kokkos_ROOT is
+  # unset.
+  if (NOT DEFINED ENV{Kokkos_ROOT})
+    set(ENV{Kokkos_ROOT} ${INSTALL_SHAREDPATH})
+  endif()
+
+  find_package(Kokkos REQUIRED)
+endif()
+
+# Albany depends on Trilinos
+if (USE_ALBANY OR USE_TRILINOS)
```

Comment:
This is fine, but for the record, I think that `USE_ALBANY` here is redundant. `find_package(ALBANY REQUIRED)` will also call `find_package(Trilinos REQUIRED)`. If a CMake package is installed well, it is responsible of finding all its TPLs when you call `find_package(YOUR_PKG)`.

Reply:
It looks like this is a deficiency with the Albany cmake system. I tried without `USE_ALBANY` check for Trilinos and I got this error:

```
CMake Error at /global/common/software/e3sm/mali_tpls/albany-e3sm-serial-release-gcc/lib64/Albany/cmake/albany-targets.cmake:61 (set_target_properties):
  The link interface of target "albanyLib" contains:

    Panzer::all_libs

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  /global/common/software/e3sm/mali_tpls/albany-e3sm-serial-release-gcc/lib64/Albany/cmake/AlbanyConfig.cmake:28 (include)
  cmake/find_dep_packages.cmake:45 (find_package)
  CMakeLists.txt:124 (include)
```

Reply:
Ah, we are probably not packaging albany that well.. :/

Reply:
I am going through our cmake stuff, and indeed we lack a lot of stuff to consider us "good cmake citizens"... I will fix it at some point.